packages/fxa-content-server/app/scripts/models/auth_brokers/oauth-webchannel-v1.js

```diff
@@ -112,8 +112,9 @@ const OAuthWebChannelBroker = OAuthRedirectAuthenticationBroker.extend({
     if (state) {
       result.state = state;
     }
-
-    return this.send(this.getCommand('OAUTH_LOGIN'), result);
+    return this._metrics.flush().then(() => {
```

Comment:
Similar to the oauth-redirect case, metrics weren't being flushed before the login message was being sent to the browser. If the browser closes the WebView before the metrics are flushed in the onload handler, then we'd lose those metrics. This ensures they are flushed before notifying the browser.